processing/src/test/java/org/apache/druid/java/util/metrics/HttpPostEmitterMonitorTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.metrics;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.emitter.core.ConcurrentTimeCounter;
+import org.apache.druid.java.util.emitter.core.HttpPostEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class HttpPostEmitterMonitorTest
+{
+  private HttpPostEmitter mockHttpPostEmitter;
+  private HttpPostEmitterMonitor monitor;
+
+  @BeforeEach
+  public void setUp()
+  {
+    mockHttpPostEmitter = mock(HttpPostEmitter.class);
+    monitor = new HttpPostEmitterMonitor("testFeed", mockHttpPostEmitter,
+            ImmutableMap.of("dimensionKey", "dimensionValue"));
+  }
+
+  @Test
+  public void testDoMonitor()
+  {
+    when(mockHttpPostEmitter.getTotalEmittedEvents()).thenReturn(100L);
+    when(mockHttpPostEmitter.getTotalDroppedBuffers()).thenReturn(10);
+    when(mockHttpPostEmitter.getTotalAllocatedBuffers()).thenReturn(20);
+    when(mockHttpPostEmitter.getTotalFailedBuffers()).thenReturn(5);
+    when(mockHttpPostEmitter.getBatchFillingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getSuccessfulSendingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getFailedSendingTimeCounter()).thenReturn(mock(ConcurrentTimeCounter.class));
+    when(mockHttpPostEmitter.getEventsToEmit()).thenReturn(200L);
+    when(mockHttpPostEmitter.getLargeEventsToEmit()).thenReturn(75L);
+    when(mockHttpPostEmitter.getBuffersToEmit()).thenReturn(30);
+    when(mockHttpPostEmitter.getBuffersToReuse()).thenReturn(15);
```

Comment:
Nit: Can re-arrange these so that `thenReturn(values)` and `thenReturn(mocks)` are grouped together.

Reply:
Done. Pushed in the next commit.